addons/l10n_id_pos/static/tests/test_pos_qris_payment.tour.js

```diff
@@ -0,0 +1,65 @@
+import * as ProductScreen from "@point_of_sale/../tests/tours/utils/product_screen_util";
+import * as PaymentScreen from "@point_of_sale/../tests/tours/utils/payment_screen_util";
+import * as Dialog from "@point_of_sale/../tests/tours/utils/dialog_util";
+import * as ReceiptScreen from "@point_of_sale/../tests/tours/utils/receipt_screen_util";
+
+import { registry } from "@web/core/registry";
+
+function isQRDisplayedinDialog() {
+    return [
+        Dialog.is({title: "QR Code"}),
+        {
+            content: "Verify QR image is displayed",
+            trigger: ".modal-content img[src^='data:image/png;base64,']",
+            run: "click",
+        },
+    ].flat();
+}
+
+function addProductandPay() {
+    return [
+        ProductScreen.addOrderline("Test Product"),
+        ProductScreen.selectedOrderlineHas("Test Product"),
+        ProductScreen.clickPayButton(),
+
+        PaymentScreen.totalIs("1,000.00"),
+        PaymentScreen.clickPaymentMethod("QRIS", true, {amount: "1,000.00"}),
+        {
+            content: "Display QR Code Payment dialog",
+            trigger: ".button.send_payment_request.highlight",
+            run: "click",
+        },
+    ].flat();
+}
+
+registry.category("web_tour.tours").add("PaymentScreenQRISPaymentFail", {
+    test: true,
+    steps: () =>
+        [
+            addProductandPay(),
+            isQRDisplayedinDialog(),
+            Dialog.confirm(),
+            Dialog.is({title: "Payment Status Update"}),
```

Comment:
```suggestion
            Dialog.is({title: "Payment Status Update"}),
            Dialog.confirm(),
```
I think it's better to check the title before clicking the confirm button.

Reply:
oops, sorry. Looks like `Payment Status Update` means a failed payment.